### PR TITLE
Reporting

### PR DESCRIPTION
**Pull request by tinamatthews1**
…he nav_order for knn from 34 to 35

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/reporting/index.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+layout: default
+title: Reporting
+nav_order: 34
+has_children: true
+has_toc: false
+---
+
+# Reporting
+
+The Kibana **Reporting** feature allows you to generate, schedule and share reports on-demand or based on an automated schedule. On-demand reports can be instantly downloaded in PNG, PDF and CSV formats, while schedule-based reports can be configured and set up for automated periodic delivery to stakeholders. You can do this either from the **Reporting** tab in left navigation panel or from within the **Dashboard**, **Visualization** and **Discover** tabs. To download, schedule and share reports, you must have the required role permissions.
```
Comment (aetter) [inline docs/reporting/index.md:11]: Active voice.

The Kibana reporting feature allows you to generate, schedule, and share reports. You can download reports on-demand in PNG, PDF, and CSV formats and schedule reports for periodic email delivery. To get started, choose **Reporting**.

"To download, schedule and share reports, you must have the required role permissions."

This seems like an entire section of these docs. What permissions do I need to have? Are there combinations of permissions that are applicable (read-only, admin, write, etc.)?

File: docs/reporting/index.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+To create a new report from the Dashboard, Visualize or Discover page, do the following:
+
+1. From the top navigation bar, choose **Reporting**. The reporting dropdown menu allows you to generate an on-demand report and download it instantly as a PDF, PNG, or CSV file. 
+2. Choose **Download PDF** or **Download PNG** to generate on-demand reports in these formats from the Dashobard or Visualize pages. For a saved search report from the Discover page, choose **Download CSV**, since this is the default file format. You can close the report generating pop-up, while the download continues asynchronously in the background. A notification pop-up confirms that your report is generated successfully. This may take a few minutes based on the size of your dashboard, visualization or saved search.
+3. To create a schedule based report from any of the above pages, choose **Create report definition**. You will be navigated to the **Create report definition** page, where the report source is preselected and the time range is what was entered within the Dashboard, Visualize or Discover page. To continue, follow the instructions from step 6 in the next section.
```
Comment (aetter) [inline docs/reporting/index.md:21]: ~~You will be navigated to the **Create report definition** page, where the report source is preselected and the time range is what was entered within the Dashboard, Visualize or Discover page. To continue, follow the instructions from step 6 in the next section.~~

Then proceed to step 6 in the next section.

File: docs/reporting/index.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+### Create a Report using the Report Definition
+
+Report definition allows you to define various settings, triggers, and delivery options for your report. To create a report definition, do the following:
```
Comment (aetter) [inline docs/reporting/index.md:25]: Let's make it clear that this section is for automated reports, periodically delivered to your inbox. No one cares about settings and triggers, etc. They care about using this process to get an email once a week.

File: docs/reporting/index.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+5. Choose your dashboard, visualization or saved search (discover data) from the dropdown menu and select a **Time range** for the report.
+6. Choose an appropriate **File format** for the report. The default file format for a saved search report from the discover page is CSV.
+7. Add a **Header** or **Footer** for the report, if required. Headers and footers are available only for dashboard or visualization reports and visible when the report is generated.
+8. Under **Report trigger**, choose the **Trigger type** as **On demand** or **Schedule based**. On demand reports are one-time and downloaded to your machine almost instantly. While schedule based reports can be set to generate periodically.  
+    1. Under **Schedule based**, choose **Recurring** or **Cron based**.
+    2. For **Recurring**, choose the **Frequency** as Daily, By interval, Weekly, or Monthly. Enter the **Request time** and **Timezone** (time at which to send the report).
+    3. For **Cron based**, enter a custom **Cron expression**. Refer [Cron expression reference](https://opendistro.github.io/for-elasticsearch-docs/docs/alerting/cron/) for more info.
```
Comment (aetter) [inline docs/reporting/index.md:37]: Refer to

File: docs/reporting/index.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+3. Under **Report settings**, enter a **Name** and an optional **Description** for your report.
+4. Choose the **Report Source** i.e. the page from where the report is generated. Reports can be generated from the **Dashboard**, **Visualize** or **Discover** pages. Dashboard and visualization reports are generated in PDF or PNG formats, while data a report (otherwise called **saved search**) from the discover page is generated in CSV format.
+5. Choose your dashboard, visualization or saved search (discover data) from the dropdown menu and select a **Time range** for the report.
+6. Choose an appropriate **File format** for the report. The default file format for a saved search report from the discover page is CSV.
+7. Add a **Header** or **Footer** for the report, if required. Headers and footers are available only for dashboard or visualization reports and visible when the report is generated.
+8. Under **Report trigger**, choose the **Trigger type** as **On demand** or **Schedule based**. On demand reports are one-time and downloaded to your machine almost instantly. While schedule based reports can be set to generate periodically.
```
Comment (aetter) [inline docs/reporting/index.md:34]: Consistently punctuate on-demand.

File: docs/reporting/index.md [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+3. To create a schedule based report from any of the above pages, choose **Create report definition**. You will be navigated to the **Create report definition** page, where the report source is preselected and the time range is what was entered within the Dashboard, Visualize or Discover page. To continue, follow the instructions from step 6 in the next section.
+
+### Create a Report using the Report Definition
+
+Report definition allows you to define various settings, triggers, and delivery options for your report. To create a report definition, do the following:
+
+1. From the left navigation panel, choose **Reporting**. In the reporting page you will see a **Reports** table and a **Report definitions** table.
```
Comment (aetter) [inline docs/reporting/index.md:27]: ~~In the reporting page you will see a **Reports** table and a **Report definitions** table.~~

File: docs/reporting/index.md [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+2. Under **Report definitions**, choose **Create**.
+3. Under **Report settings**, enter a **Name** and an optional **Description** for your report.
+4. Choose the **Report Source** i.e. the page from where the report is generated. Reports can be generated from the **Dashboard**, **Visualize** or **Discover** pages. Dashboard and visualization reports are generated in PDF or PNG formats, while data a report (otherwise called **saved search**) from the discover page is generated in CSV format.
```
Comment (aetter) [inline docs/reporting/index.md:30]: while data a report?

File: docs/reporting/index.md [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+8. Under **Report trigger**, choose the **Trigger type** as **On demand** or **Schedule based**. On demand reports are one-time and downloaded to your machine almost instantly. While schedule based reports can be set to generate periodically.  
+    1. Under **Schedule based**, choose **Recurring** or **Cron based**.
+    2. For **Recurring**, choose the **Frequency** as Daily, By interval, Weekly, or Monthly. Enter the **Request time** and **Timezone** (time at which to send the report).
+    3. For **Cron based**, enter a custom **Cron expression**. Refer [Cron expression reference](https://opendistro.github.io/for-elasticsearch-docs/docs/alerting/cron/) for more info.
+9. For **Delivery settings**, choose **Email recipients**. 
+10. For **Email configuration** select or add email recipients. 
+11. Select an **Email format**.  It can be sent either as an embedded HTML report, or as an attachment. Also add an **Email subject** and an optional message for the email.
+12. Choose **Create**. You will be navigated to the **Reporting** landing page where you can view your report.
```
Comment (aetter) [inline docs/reporting/index.md:41]: No future tense. `Kibana redirects you to the **Reporting** page` or something.

**Review by aetter — CHANGES_REQUESTED** [review]

File: docs/reporting/index.md (hunk 2 above)
Comment (tinamatthews1) [reply docs/reporting/index.md:21]: done

File: docs/reporting/index.md (hunk 3 above)
Comment (tinamatthews1) [reply docs/reporting/index.md:25]: done

File: docs/reporting/index.md (hunk 4 above)
Comment (tinamatthews1) [reply docs/reporting/index.md:37]: done

File: docs/reporting/index.md (hunk 5 above)
Comment (tinamatthews1) [reply docs/reporting/index.md:34]: done

File: docs/reporting/index.md (hunk 6 above)
Comment (tinamatthews1) [reply docs/reporting/index.md:27]: done

File: docs/reporting/index.md (hunk 7 above)
Comment (tinamatthews1) [reply docs/reporting/index.md:30]: corrected it, sorry. It was supposed to be - "while data reports are .."

File: docs/reporting/index.md (hunk 8 above)
Comment (tinamatthews1) [reply docs/reporting/index.md:41]: done

File: docs/reporting/index.md (hunk 1 above)
Comment (tinamatthews1) [reply docs/reporting/index.md:11]: I corrected the first part. For role permissions, they're working on integrating this plug-in with the security plug-in. I'm not sure what the details are yet. Maybe Ashwin and follow-up and have this completed. It is due for release only in January.

**Pull request closed without merging**